omfwd: inconsistent use of ATOMIC_MUT versus ATOMIC_MUT64 leads to compilation errors

omfwd fails to build when HAVE_ATOMIC_BUILTINS is defined but HAVE_ATOMIC64_BUILTINS is not defined.  The compilation error is due to inconsistent use of macros against the member mut_nActiveTargets.

It is defined using

    DEF_ATOMIC_HELPER_MUT(mut_nActiveTargets);

However it is inconsistently accessed using both MUT and MUT64 macros

```
    INIT_ATOMIC_HELPER_MUT64(pData->mut_nActiveTargets);
    DESTROY_ATOMIC_HELPER_MUT(pData->mut_nActiveTargets);
```

In this case, with HAVE_ATOMIC64_BUILTINS not being defined, the INIT_ATOMIC_HELPER_MUT64 attempts to initialize the member variable, however because HAVE_ATOMIC_BUILTINS is defined, that member variable was never created.

I suspect in this environment (bitbake x86_64 host cross-compiling to x86_64 OpenEmbedded), the ATOMIC64 was explicitly turned off by configure, since there is a presumption that the compiled code would not be able to run in the host environment.

```
  if test "$cross_compiling" = yes; then :
  ap_cv_atomic_builtins_64=no
```

Normally, the cross_compiling check would also apply to HAVE_ATOMIC_BUILTINS, but in this case configure shows it cached, as it seems to have been passed explicitly to configure.

```
checking whether the compiler provides atomic builtins... (cached) yes
checking whether the compiler provides atomic builtins for 64 bit data types... no
```

A workaround seems to pass the ap_cv_atomic64_builtins_64=yes when running configure, however the underlying code should still be changed to consistently use either the MUT version or the MUT64 version of the macros.

```
| ../../rsyslog-8.2504.0.master/tools/omfwd.c:478:34: error: no member named 'mut_nActiveTargets' in 'struct _instanceData'; did you mean 'nActiveTargets'?
|         INIT_ATOMIC_HELPER_MUT64(pData->mut_nActiveTargets);
|                                         ^~~~~~~~~~~~~~~~~~
|                                         nActiveTargets
| ../../rsyslog-8.2504.0.master/runtime/atomic.h:261:59: note: expanded from macro 'INIT_ATOMIC_HELPER_MUT64'
```

--- a/tools/omfwd.c
+++ b/tools/omfwd.c
@@ -835,7 +835,7 @@ BEGINcreateInstance
     /* We always have at least one target and port */
     pData->nTargets = 1;
     pData->nActiveTargets = 0;
-    INIT_ATOMIC_HELPER_MUT64(pData->mut_nActiveTargets);
+    INIT_ATOMIC_HELPER_MUT(pData->mut_nActiveTargets);
     pData->nPorts = 1;
     pData->target_name = NULL;
     if (cs.pszStrmDrvr != NULL) CHKmalloc(pData->pszStrmDrvr = (uchar *)strdup((char *)cs.pszStrmDrvr));

